src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -0,0 +1,68 @@
+﻿//import sql from "mssql";
+const sql = require('mssql');
+import * as assert from "assert";
+import { IDataProvider } from "../IDataProvider";
+import { ConnectionPool, ConnectionError, config, IResult } from "mssql";
+import { SeasonInfo } from "../../DataModels/SeasonInfo";
+
+/*
+ * Sql Data Provider
+ */
+export class SqlDataProvider implements IDataProvider {
+    private _tablePrefix: string = "Hymns_";
+
+    constructor(private _config: config) {
+    }
+
+    private _getQualifiedName(sp: string): string {
+        return `${this._tablePrefix}${sp}`
+    }
+
+    //TODO: why is the return type requiring void?
+    public getSeasonList(): Promise<void | SeasonInfo[]> {
+        const cp = new ConnectionPool(this._config);
```

Comment:
I *think* (though not 100% sure) you're supposed to re-use a connection pool across your application. see [here](https://www.npmjs.com/package/mssql#connection-pools)